frontend/src/metabase/querying/viz-settings/utils/sync-viz-settings.ts

```diff
@@ -232,7 +232,7 @@ function syncGraphMetrics(
   return {
     ...settings,
     "graph.metrics": syncColumnNames({
-      settings: graphMetrics,
+      settings: graphMetrics.filter((metric): metric is string => !!metric),
```

Comment:
The reason you needed this is because of the change introduced in https://github.com/metabase/metabase/pull/49414. I'd rather leave the code alone, and fix the type instead.

That PR says `settings` is `string[]`, would likely be because of `"graph.metrics"?: string[];`. I'd change the `settings` type in `SyncColumnNamesOpts` to be `(string | null)[]` instead. There would be a series of type errors that you need to fix though, but I'd rather fix the type than messing with the value and potentially introduce bugs we do not foresee.

Reply:
very fair, i'd much rather deal with type errors than actual errors

Reply:
Agree with Kelvin. The viz settings land is full of specific decisions that are not documented well enough. I suggest not changing the actual behavior even slightly since the goal if this PR is to fix types

Reply:
You could use `isNotNull` here:
```suggestion
      settings: graphMetrics.filter(isNotNull),
```